eww: 0.6.0 -> 0.6.0-unstable-2024-04-26

JohnRTitor opened the pull request:
## Description of changes
This PR merely backports the upstream patch (https://github.com/elkowar/eww/commit/2c8811512460ce6cc75e021d8d081813647699dc) to fix the problem in current release.
Also reformats the file using `nixfmt-rfc-style`.
Fixes https://github.com/NixOS/nixpkgs/issues/308473

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).


---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

eclairevoyant commented on pkgs/by-name/ew/eww/package.nix:
```diff
@@ -32,18 +46,27 @@ rustPlatform.buildRustPackage rec {
 
   meta = {
-    description = "ElKowars wacky widgets";
+    description = "Elkowars Wacky Widgets, a widget system made in Rust to allow you create widgets for any WM.";
```
This can go in longDescription but not description, this is a bit long and doesn't follow guidelines (shouldn't end in period). Alternatively you can rephrase this to be shorter

JohnRTitor replied:
Rephrased the description to be shorter
Moved the long stuff to longDescription, does this look fine now?

eclairevoyant commented on pkgs/by-name/ew/eww/package.nix:
```diff
@@ -17,13 +19,25 @@ rustPlatform.buildRustPackage rec {
     owner = "elkowar";
     repo = "eww";
-    rev = "refs/tags/v${version}";
+    rev = "v${version}";
```
remove this change

JohnRTitor replied:
Can you consider explaining why? fetchFromGitHub can find the tag just fine if I leave it like this.

eclairevoyant replied:
There's absolutely no reason for this change. And it's better to be explicit especially if branches get created in the future as well as for the sake of update scripts.

JohnRTitor replied:
Reverted, thanks.

eclairevoyant commented on pkgs/by-name/ew/eww/package.nix:
```diff
@@ -17,13 +19,25 @@ rustPlatform.buildRustPackage rec {
   };
 
+  patches = [
+    # FIXME: remove with the next update
```
Would be best to also explain the patch's purpose, or at least link to the issue

JohnRTitor replied:
Done.

eclairevoyant reviewed (changes requested):
Thanks for the PR. Some changes below:

w-lfchen commented on pkgs/by-name/ew/eww/package.nix:
```diff
@@ -21,9 +23,22 @@ rustPlatform.buildRustPackage rec {
     hash = "sha256-rzDnplFJNiHe+XbxbhZMEhPJMiJsmdVqtZxlxhzzpTk=";
   };
 
+  patches = [
+    # FIXME: remove with the next release
+    # https://github.com/elkowar/eww/pull/1084 - patch to fix string truncation in eww config
+    (fetchpatch {
+      name = "fix-eww-string-truncation.patch";
+      url = "https://github.com/elkowar/eww/commit/2c8811512460ce6cc75e021d8d081813647699dc.patch";
+      hash = "sha256-oYrpuTI8KbMtboOdc8xw3UrAEkLu5xnphgey7koKmrc=";
+    })
+  ];
```
Is there a reason for not just changing the source ref to the commit you're fetching here?

JohnRTitor replied:
I thought the same, since it also [incorporates](https://github.com/elkowar/eww/compare/v0.6.0...2c8811512460ce6cc75e021d8d081813647699dc) Cargo.toml and other changes I was hesitant.
Second opinion, @eclairevoyant?

w-lfchen replied:
I'd be really surprised if those changes cause issues

JohnRTitor replied:
Yep, thought the same :) looked at the changes, pretty harmless bugfixes and deps update

eclairevoyant replied:
Either is fine, though if you're changing the commit, you'll also have to change the version to the unstable format.

JohnRTitor replied:
Using the version `0.6.0-unstable-2024-04-26`.

w-lfchen commented on pkgs/by-name/ew/eww/package.nix:
```diff
@@ -32,18 +47,34 @@ rustPlatform.buildRustPackage rec {
+      Eww (ElKowar's Wacky Widgets) is a widget system made in Rust, which lets
+      you create your own widgets similarly to how you can in AwesomeWM.
+      The key difference: It is independent of your window manager!
+      It can be configured in yuck and themed using CSS, very easy
```
I'd change this to "... using CSS, **is** very easy ..."

JohnRTitor replied:
Done.

w-lfchen commented on pkgs/by-name/ew/eww/package.nix:
```diff
@@ -32,18 +47,34 @@ rustPlatform.buildRustPackage rec {
-    description = "ElKowars wacky widgets";
+    description = "A widget system made in Rust to create widgets for any WM";
+    longDescription = ''
+      Eww (ElKowar's Wacky Widgets) is a widget system made in Rust, which lets
```
This comma feels wrong to me but I am no English native speaker so I could be mistaken

JohnRTitor replied:
Well, I've always used it that way :)
better to have another opinion

eclairevoyant replied:
I agree with both of w-lfchen's suggestions about the description.

JohnRTitor replied:
Done.

JohnRTitor committed:
eww: 0.6.0 -> 0.6.0-unstable-2024-04-26
use 2c8811512460ce6cc75e021d8d081813647699dc to fix string truncation bug
https://github.com/elkowar/eww/pull/1084

JohnRTitor committed:
eww: update description, add long description
reformat using nixfmt-rfc-style

w-lfchen reviewed (approved):
LGTM
